### DISCUSSION TITLE
How to make the home page layout like a sidebar + docs page?

### DISCUSSION THREAD

**Discussion by davestewart** [vuejs/vitepress#4373] [Q&A] [answered]
For our internal docs, we are using the default theme, but we don't need a splashy home page, and would instead like to make the home page behave like a docs page.

We know we can remove the frontmatter and have it display like `layout: doc` but the problem is the `container`, `content` and `VPNavBar`  have different widths, padding, etc to the main docs pages, so when you navigate, there's a big layout jump:

```
+-------------------------+
|         :        :      |
|         :  Home  :      |
|         :        :      |
|         :        :      |
|         :        :      |
+-------------------------+
```
```
+-------+-----------------+
|       |            :    |
|       |  Content   :    |
|       |            :    |
|       |            :    |
|       |            :    |
+-------+-----------------+
```

I've tried mucking about with custom styles, but there's a lot to be considered, especially with breakpoints, etc.

I also tried rewrites (to skip the home page and jump directly to a section) but I think I need to those on the server level, and this is just internal.

Does anyone know how we can make the home page behave more consistently?

Thanks.

**Comment by brc-dd** [comment]
What does your sidebar config look like? 

I’m doing this at my company — https://sefirot.globalbrains.com/ — we just have a index.md in docs without home layout — https://github.com/globalbrain/sefirot/tree/main/docs

**Comment by davestewart** [comment]
Ah, I'm running the "multiple sidebars" setup with a navigation bar.

Home page:

![CleanShot 2024-11-20 at 22 32 26](https://github.com/user-attachments/assets/2261431f-4b4d-454a-9187-b5b67a9390ea)

Section page:

![CleanShot 2024-11-20 at 22 32 42](https://github.com/user-attachments/assets/cbdab199-0f0f-4258-8079-28c47913f93f)

Config:

```ts
import { defineConfig } from 'vitepress'
import { withSidebar } from 'vitepress-sidebar'

// -----------------------------------------------------------------------------
// helpers
// -----------------------------------------------------------------------------

// @see https://vitepress-sidebar.cdget.com
function getMenu(folder) {
  return {
    documentRootPath: `docs`,
    scanStartPath: folder,
    basePath: `/${folder}/`,
    resolvePath: `/${folder}/`,
    useTitleFromFileHeading: true,
    useFolderTitleFromIndexFile: true,
    capitalizeEachWords: true,
    includeFolderIndexFile: true,
  }
}

...

export default withSidebar(config, [
  getMenu('setup'),
  getMenu('backend'),
  getMenu('frontend'),
  getMenu('specs'),
])
```

**Reply by brc-dd** [reply]
Is there something you can show on sidebar on your home page? Or can you add home page to setup sidebar maybe? Something like this - https://stackblitz.com/edit/vite-sy1hhp?file=docs/.vitepress/config.ts

**Reply by davestewart** [reply]
That's a nice idea for a workaround!

Initially I thought it wasn't going to work as it looks pretty weird having just one sidebar item.

BUT... enhance app to the rescue!

```ts
// theme/index.ts
  ...
  enhanceApp({ app, router, siteData }) {
    router.onAfterRouteChanged = (to)  => {
      document.querySelector('#app').setAttribute('data-route', to)
    }
  },
```

``` css
// custom.css
#app[data-route="/"] {
  --vp-sidebar-bg-color: none;

  #VPSidebarNav {
    display: none;
  }

  .VPNavBarTitle .title {
    border-bottom-color: transparent;
  }
}
```

The docs look great now; a simple home page and no layout shift when I navigate.

I do think it would be cool if the sidebar layout had a way to be more consistent, but I'll take this!

**Reply by brc-dd** [reply]
You can actually add class without hack too:

```md
---
pageClass: 'home'
---

markdown here...
```

```css
:root.home {
  ...
}
```

EDIT (06-08-2025): this doesn't work. I thought this was being added to root (`<html>` tag). But it's added only to layout.

**Reply by davestewart** [reply, marked as answer]
OK, this is now the final solution...

I used a `null` in the sidebar config, which is cleaner:

```js
sidebar: {
  '/': [null],
  ...
}
```

And I kept the router "hack" as it just means the logic to hide the sidebar is in one place (as above).

The router hook is now:

```ts
  enhanceApp({ app, router, siteData }) {
    router.onAfterRouteChanged = (to)  => {
      const sidebar = siteData.value.themeConfig?.sidebar[to]
      const noSidebar = sidebar?.length === 1 && sidebar[0] === null
      document.querySelector('#app')?.classList.toggle('noSidebar', noSidebar)
    }
  },
```

The CSS is now:

```css
#app.noSidebar {
  --vp-sidebar-bg-color: none;

  #VPSidebarNav {
    display: none;
  }

  .VPNavBarTitle .title {
    border-bottom-color: transparent;
  }
}
```

Demo here:

- [https://stackblitz.com/edit/vite-hidden-sidebar](https://stackblitz.com/edit/vite-hidden-sidebar?file=docs%2F.vitepress%2Ftheme%2Findex.ts,docs%2F.vitepress%2Ftheme%2Fcustom.css,docs%2F.vitepress%2Fconfig.ts,docs%2Findex.md)
